applications/GeoMechanicsApplication/custom_elements/geo_steady_state_Pw_piping_element.h

```diff
@@ -0,0 +1,298 @@
+        }
+    }
+
+    static void AddContributionsToLhsMatrix(MatrixType& rLeftHandSideMatrix,
+                                            const BoundedMatrix<double, TNumNodes, TNumNodes>& rPermeabilityMatrix)
+    {
+        rLeftHandSideMatrix = rPermeabilityMatrix;
+    }
+
+    void AddContributionsToRhsVector(VectorType& rRightHandSideVector,
+                                     const BoundedMatrix<double, TNumNodes, TNumNodes>& rPermeabilityMatrix,
+                                     const array_1d<double, TNumNodes>& rFluidBodyVector) const
+    {
+        const auto permeability_vector =
+            array_1d<double, TNumNodes>{-prod(rPermeabilityMatrix, GetNodalValuesOf(WATER_PRESSURE))};
+        rRightHandSideVector = permeability_vector + rFluidBodyVector;
+    }
+
+    Vector CalculateIntegrationCoefficients(const Vector& rDetJContainer) const
+    {
+        const auto& r_properties         = GetProperties();
+        const auto& r_integration_points = GetGeometry().IntegrationPoints(GetIntegrationMethod());
+
+        auto result = Vector{r_integration_points.size()};
+        // all governed by PIPE_HEIGHT and element length so without CROSS_AREA
+        std::transform(r_integration_points.begin(), r_integration_points.end(), rDetJContainer.begin(),
+                       result.begin(), [](const auto& rIntegrationPoint, const auto& rDetJ) {
+            return rIntegrationPoint.Weight() * rDetJ;
+        });
+        return result;
+    }
+
+    Matrix FillPermeabilityMatrix(double pipe_height) const
+    {
+        Matrix constitutive_matrix(1, 1);
+        constitutive_matrix(0, 0) = std::pow(pipe_height, 3) / 12.0;
+        return constitutive_matrix;
+    }
+
+    BoundedMatrix<double, TNumNodes, TNumNodes> CalculatePermeabilityMatrix(
+        const GeometryType::ShapeFunctionsGradientsType& rShapeFunctionGradients,
+        const Vector&                                    rIntegrationCoefficients) const
+    {
+        RetentionLaw::Parameters RetentionParameters(GetProperties());
+        const auto&              r_properties              = GetProperties();
+        const double             dynamic_viscosity_inverse = 1.0 / r_properties[DYNAMIC_VISCOSITY];
+
+        auto constitutive_matrix = FillPermeabilityMatrix(r_properties[PIPE_HEIGHT]);
+
+        auto result = BoundedMatrix<double, TNumNodes, TNumNodes>{ZeroMatrix{TNumNodes, TNumNodes}};
+        for (unsigned int integration_point_index = 0;
+             integration_point_index < GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
+             ++integration_point_index) {
+            result += GeoTransportEquationUtilities::CalculatePermeabilityMatrix<TDim, TNumNodes>(
+                rShapeFunctionGradients[integration_point_index], dynamic_viscosity_inverse,
+                constitutive_matrix, 1.0, rIntegrationCoefficients[integration_point_index]);
+        }
+
+        return result;
+    }
```

Comment:
In the experiment branch with calculators (see #12763), the calculation of the permeability matrix is delegated to these new calculators. Here, the calculation is of course slightly different. In the longer term, should we have a version of the permeability calculator that does it slightly different for piping vs 'normal', or should we handle that differently? 

Looping @avdg81